buidHostLin.md improvements

andrej opened the pull request:
I made these edits after following the instructions on a completely fresh install of Ubuntu 22.04.

1. Reformat.
2. Rewrite and give rationale for steps in a couple places for better user experience.
3. Include steps that were previously only referenced by links. I think it makes more sense to include them so that a user can simply follow a linear list of steps.
4. Add a couple missing prerequisites.
5. Rewrite code so that it should mostly just work when users blindly copy and paste it in. This should be fairly close to a script that just does all of this.

jgmelber commented:
LGTM. If possible propagate any improvements for clarity to the Windows 11 getting started guide.

andrej commented:
I've refactored the Windows instructions a little bit and added a troubleshooting section to the Linux to cover some issues I had.

Please merge if it looks okay, as I do not have the rights to merge
